workers/unity/Packages/com.improbable.gdk.playerlifecycle/PlayerLifecycleHelper.cs

```diff
@@ -21,6 +23,47 @@ public static class PlayerLifecycleHelper
             template.AddComponent(owningComponent, serverAccess);
         }
 
+        public static bool SerializeArguments(Object playerCreationArguments, out byte[] serializedArguments)
+        {
+            try
+            {
+                using (var memoryStream = new MemoryStream())
+                {
+                    var binaryFormatter = new BinaryFormatter();
+                    binaryFormatter.Serialize(memoryStream, playerCreationArguments);
+                    serializedArguments = memoryStream.ToArray();
+                    return true;
+                }
+            }
+            catch (Exception e)
+            {
+                UnityEngine.Debug.LogError($"Unable to serialize player creation arguments. {e.Message}");
+                serializedArguments = null;
+                return false;
+            }
+        }
+
+        public static bool DeserializeArguments<T>(byte[] serializedArguments, out T deserializedArguments)
+        {
+            try
+            {
+                using (var memoryStream = new MemoryStream())
+                {
+                    var binaryFormatter = new BinaryFormatter();
+                    memoryStream.Write(serializedArguments, 0, serializedArguments.Length);
+                    memoryStream.Seek(0, SeekOrigin.Begin);
+                    deserializedArguments = (T) binaryFormatter.Deserialize(memoryStream);
+                    return true;
+                }
+            }
+            catch (Exception e)
+            {
+                UnityEngine.Debug.LogError($"Unable to deserialize player creation arguments. {e.Message}");
```

Comment:
this should probably throw an exception as there is no way to recover from this

Reply:
I think it should be up to the caller to do this. We should just have byte[] as an argument and leave it at that. If we want to do this I would much rather make it structured data though and provide a way of making this a schema type.

Reply:
Agree with @samiwh 100%